File: frontend/src/components/applying/dto/ApplyingResponseDTO.ts
import { PostingResponseDTO } from "../../posting/dto";

export interface BaseApplyingDTO {
    applyingId?: number;
    postingId?: number;
    userId: number;
    nickname: string;
    prContent: string;
    status: number; // 0: 대기, 1: 승인, -1: 거절
    statusText: string;
    applyDatetime: string;
    confirmationDatetime: string | null;
    posting?: PostingResponseDTO;
    
    // CommonApplyResponseDTO와 동일하게 추가
    applyHistoryId?: number;
    postHistoryId?: number;
}

export interface ApplyingResponseDTO extends BaseApplyingDTO {
}

export interface MyApplyingResponseDTO extends BaseApplyingDTO {
    // BaseApplyingDTO에 이미 포함되어 있음
}

File: src/main/java/wedding/alba/function/applying/dto/ApplyingResponseDTO.java
package wedding.alba.function.applying.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import wedding.alba.function.posting.dto.PostingResponseDTO;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ApplyingResponseDTO {
    private Long applyingId;
    private Long postingId;
    private PostingResponseDTO posting;
    private Long userId;
    private String nickname;
    private String prContent;
    private Integer status;
    private String statusText;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime applyDatetime;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime confirmationDatetime;

    public void setStatusStr() {
        if(this.status == 0) {
            this.statusText = "대기";
        } else if(this.status == 1) {
            this.statusText = "승인";
        } else if(this.status == -1) {
            this.statusText = "거절";
        } else {
            this.statusText = "알수없음";
        }
    }

}
